app/src/ui/dispatcher/dispatcher.ts

```diff
@@ -2522,15 +2523,49 @@ export class Dispatcher {
    */
   public async rerequestCheckSuites(
     repository: GitHubRepository,
-    checkRuns: ReadonlyArray<IRefCheck>
+    checkRuns: ReadonlyArray<IRefCheck>,
+    failedOnly: boolean
   ): Promise<ReadonlyArray<boolean>> {
-    // Get unique set of check suite ids
-    const checkSuiteIds = new Set<number | null>([
-      ...checkRuns.map(cr => cr.checkSuiteId),
-    ])
-
     const promises = new Array<Promise<boolean>>()
 
+    // If it is one and in actions check, we can rerun it individually.
+    if (
+      checkRuns.length === 1 &&
+      checkRuns[0].actionsWorkflow !== undefined &&
+      enableReRunFailedAndSingleCheckJobs()
+    ) {
+      promises.push(
+        this.commitStatusStore.rerunJob(repository, checkRuns[0].id)
+      )
+      return Promise.all(promises)
+    }
+
+    const checkSuiteIds = new Set<number>()
+    const workflowRunIds = new Set<number>()
+    for (const cr of checkRuns) {
+      if (
+        failedOnly &&
+        cr.actionsWorkflow !== undefined &&
+        enableReRunFailedAndSingleCheckJobs()
+      ) {
+        workflowRunIds.add(cr.actionsWorkflow.id)
+        continue
+      }
+
+      // There could still be failed ones that are not action and only way to
+      // rerun them it rerun their whole check suite
+      if (cr.checkSuiteId !== null) {
+        checkSuiteIds.add(cr.checkSuiteId)
+      }
+    }
+
+    for (const id of workflowRunIds) {
+      if (id === null) {
```

Comment:
According to the type of `workflowRunIds` (`Set<number>`). This can't ever be null. If it can be null we need to update the type of `id` in `IAPIWorkflowRun` and update the set accordingly.

Reply:
Shouldn't be able to be null -> Left over from previous solution and didn't realize new solution didn't need it.